server/src/main/java/org/elasticsearch/index/engine/Engine.java

```diff
@@ -214,6 +214,25 @@ public void verifyEngineBeforeIndexClosing() throws IllegalStateException {
         }
     }
 
+    public interface IndexCommitListener {
+
+        /**
+         * This method is invoked each time a new Lucene commit is created through this engine. The {@link IndexCommitRef} prevents the
+         * {@link IndexCommitRef} files to be deleted from disk until the reference is closed. As such, the listener must close the
+         * reference as soon as it is done with it.
+         *
+         * @param indexCommitRef a reference on the newly created index commit
+         */
+        void onNewCommit(ShardId shardId, Engine.IndexCommitRef indexCommitRef);
+
+        /**
+         * This method is invoked after the policy deleted the given {@link IndexCommit}.
```

Comment:
Can we add that it will never be called on a commit before the corresponding `indexCommitRef` received through `onNewCommit` has been closed, but could be called directly during such a `close()`?